src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java

```diff
@@ -143,28 +144,26 @@ public FailoverConnectionPlugin(
       Log logger) {
     this.currentConnectionProvider = currentConnectionProvider;
     this.propertySet = propertySet;
-    this.mainHost = this.currentConnectionProvider.getCurrentHostInfo();
     this.nextPlugin = nextPlugin;
     this.logger = logger;
-    this.initialConnectionProps = this.mainHost.getHostProperties();
-
-    initSettings();
-    this.connectionProvider = new BasicConnectionProvider();
 
-    if (!this.enableFailoverSetting) {
-      // Use a standard default connection - no further initialization required
-      try {
-        this.updateCurrentConnection(
-            this.connectionProvider.connect(this.mainHost),
-            this.mainHost);
-      } catch (SQLException e) {
-        // todo: proper handling
-        e.printStackTrace();
-      }
+    this.initialConnectionProps = new HashMap<>();
+    Properties originalProperties = this.propertySet.exposeAsProperties();
+    if (originalProperties != null) {
+      originalProperties.stringPropertyNames().stream()
+        .forEach(p -> this.initialConnectionProps.put(p, originalProperties.getProperty(p)));
+    }
```

Comment:
```suggestion
      for (String p : originalProperties.stringPropertyNames()) {
        this.initialConnectionProps.put(p, originalProperties.getProperty(p));
      }
```

Reply:
`originalProperties .forEach((key, value) -> this. initialConnectionProps.put((String) key, (String) value));` ?

Reply:
We need to copy the additional properties from the host info
```
    Properties additionalProperties = this.currentConnectionProvider.getCurrentHostInfo().exposeAsProperties();

    for (String p : additionalProperties.stringPropertyNames()) {
      if (!this.initialConnectionProps.containsKey(p)) {
        this.initialConnectionProps.put(p, additionalProperties.getProperty(p));
      }
    }
```